### PR TITLE
Use the resolved revision instead of the provided git treeish when choosing API baseline location

### PR DESCRIPTION
**Pull request by owenv**
Change the naming of API baseline directories to use a resolved revision instead of the trees provided as input to experimental-api-diff

### Motivation:

Currently, when comparing the api of a package to a branch, it's possible to end up with a stale baseline if that branch is updated and the baseline isn't manually regenerated. 

### Modifications:

This PR uses the resolved revision instead, which captures the current state of the branch/tag when the baseline is generated. Comparisons to a specific commit are unchanged

### Result:

The bug described above should be fixed

### CODE REVIEW

**Comment by owenv** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

**Review by elsh — APPROVED** [review]

File: Tests/CommandsTests/APIDiffTests.swift [hunk 1]
```diff
@@ -317,7 +318,46 @@ final class APIDiffTests: XCTestCase {
                     XCTFail("Unexpected error")
                     return
                 }
-                XCTAssertTrue(stderr.contains("error: Couldn’t check out revision ‘7.8.9’"))
+                print(stderr)
```
Comment (elsh) [inline Tests/CommandsTests/APIDiffTests.swift:321]: Is this for debugging?

Comment (owenv) [reply Tests/CommandsTests/APIDiffTests.swift]: yeah, I left that in by accident, thanks!

**Comment by owenv** [comment]
@swift-ci please smoke test

**Comment by owenv** [comment]
@swift-ci please smoke test Linux

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by elsh** [comment]
@owenv  https://github.com/apple/swift-package-manager/pull/3633 is merged. Is there anything that needs to be merged from this PR? If not, feel free to close.

**Comment by owenv** [comment]
> @owenv #3633 is merged. Is there anything that needs to be merged from this PR? If not, feel free to close.

Thanks - all the changes in this PR were merged there, closing

**Pull request closed without merging**